sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp

```diff
@@ -9,7 +9,7 @@
 #include <string>
 #include <vector>
 
-#include <azure/core/strings.hpp>
+#include <azure/core/internal/strings.hpp>
```

Comment:
let's create an issue here about moving this to a cpp file. Since this hpp should not be including internal header

Reply:
@Jinming-Hu  @katmsft https://github.com/Azure/azure-sdk-for-cpp/issues/1376  FYI

Reply:
I thought that was only true for `private` headers.

Reply:
Well, any end-user including storage_common.hpp  (or some other header that includes this header) would create a dependency to this core-internal api. Hopefully nothing ever happen as long as the end-user never uses the header and the dependency is not strongly coupled.
So, I consider a better practice to use private and internal headers from cpp files only.

Reply:
> let's create an issue here about moving this to a cpp file.


> Since this hpp should not be including internal header

Why not? We put things in `Internal` namespace to make it less visible to end users. Now we cannot even include the header at all?